File: 250714/두 방향 탈출 가능 여부 판별하기/determine-escapableness-with-2-ways.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const grid = input.slice(1, n + 1).map(row => row.split(' ').map(Number));
const visited = Array.from({length : n}, () => Array(m).fill(false))
const dx = [1, 0];
const dy = [0, 1];

function dfs(x, y){
    for(let i = 0; i < 2; i++){
        const nx = x + dx[i];
        const ny = y + dy[i];

        if(0 <= ny && ny < n && 0 <= nx && nx < m  && !visited[ny][nx] && grid[ny][nx] === 1){
            visited[ny][nx] = true
            dfs(nx, ny)
        }
    }
}

dfs(0, 0);

console.log(visited[n-1][m-1] ? 1 : 0)

File: 250820/두 방향 탈출 가능 여부 판별하기/determine-escapableness-with-2-ways.py
n, m = map(int, input().split())
grid = [list(map(int, input().split())) for _ in range(n)]
visited = [[False for _ in range(m)] for _ in range(n)]


dx = [1, 0]
dy = [0, 1]

def dfs(x, y):
    for i in range(2):
        curX = dx[i] + x
        curY = dy[i] + y
        if 0 <= curX and curX < m and 0<= curY \
        and curY < n and grid[curY][curX] == 1 \
        and not visited[curY][curX]:
            visited[curY][curX] = True
            dfs(curX, curY)


visited[0][0] = True
dfs(0, 0)

print(1 if visited[n-1][m-1] else 0)
